mbuild/lattice.py

```diff
@@ -0,0 +1,260 @@
+                             'Please use 1, 2, or 3 for the dimension.'
+                             .format(dimension))
+        else:
+            dimension = int(dimension)
+
+        """
+        Cleaning up the lattice_vectors input.
+        If the input is not provided, assume unit vectors along x,y,z or
+        x,y if 2D.
+        Check for the user to input a list or a numpy array for the
+        lattice_vectors.
+        If user inputted, type will be converted to float, and
+        dimensionalty will be enforced. Must define all vectors for
+        implemented dimension (3x3 for 3D, 2x2 for 2D).
+        Will check user implemented for co-linear vectors, these cannot
+        define a space filling unit cell.
+        Also checking for right-handed lattice_vectors, their
+        determinant must be > 0.
+        """
+        if lattice_vectors is None:
+            if dimension is 3:
+                lattice_vectors = np.asarray(([1.0, 0.0, 0.0],
+                                              [0.0, 1.0, 0.0],
+                                              [0.0, 0.0, 1.0]))
+            elif dimension is 2:
+                lattice_vectors = np.asarray(([1.0, 0.0], [0.0, 1.0]))
+            else:
+                lattice_vectors = np.asarray(([1.0]))
+        else:
+            lattice_vectors = np.asarray(lattice_vectors, dtype=float)
+            shape = np.shape(lattice_vectors)
+            if dimension is 3:
+                if (3, 3) != shape:
+                    raise ValueError('Dimensionality of'
+                                     'lattice_vectors is of {},'
+                                     ' not (3,3).' .format(shape))
+                volume = np.linalg.det(lattice_vectors)
+                if abs(volume) == 0.0:
+                    raise ValueError('Co-linear vectors {}'
+                                     'have a volume of 0.0.'
+                                     'Does not define a unit cell.'
+                                     .format(lattice_vectors))
+                if volume < 0.0:
+                    raise ValueError('Negative Determinant: '
+                                     'the volume of {} is '
+                                     'negative, indicating a '
+                                     'left-handed system.'
+                                     .format(volume))
+            elif dimension is 2:
+                if (2, 2) != shape:
+                    raise ValueError('Dimensionality of lattice_vectors is'
+                                     ' of {}, not (2,2).' .format(shape))
+                area = np.linalg.det(lattice_vectors)
+                if abs(area) == 0.0:
+                    raise ValueError('Co-linear vectors {}'
+                                     'have an area of 0.0. Does not '
+                                     'define a unit cell.'
+                                     .format(lattice_vectors))
+                if area < 0.0:
```

Comment:
You can remove some of the duplicate volume/area code here by just using the same variable and doing the checks after the if dim==3 elif dim ==2 block